addon/edit/matchbrackets.js

```diff
@@ -106,13 +121,15 @@
       cm.off("cursorActivity", doMatchBrackets);
     if (val) {
       cm.state.matchBrackets = typeof val == "object" ? val : {};
+      cm.state.matchBrackets.fatCursor = 
+          cm.getWrapperElement().classList.contains('cm-fat-cursor');
```

Comment:
done * 3.  I wasn't sure if checking the DOM every time the cursor moves was kosher, but maybe it's ok